src/main/java/net/mcreator/element/types/Item.java

```diff
@@ -194,8 +194,8 @@ public LinkedHashMap<String, ModelEntry> filterModels() {
 		modelsMap.forEach((state, model) -> {
 			List<String> states = new ArrayList<>();
 			for (String match : state.split(",")) {
-				String property = match.split("=")[0];
-				if (customProperties.containsKey(property) || builtinProperties.contains(property))
+				String prop = match.split("=")[0];
+				if (customProperties.containsKey(prop.replace("CUSTOM:", "")) || builtinProperties.contains(prop))

```

Comment:
You still remove CUSTOM: before it reaches freemarker. CUSTOM: removal will be done by the mapping system for you

Reply:
(I literally opened the pull request page the moment you reviewed it :D)
This is only for filtering properties, it does not modify the map contents (as you can see on the next line).